kpi/tests/api/v2/test_api_asset_delete.py

```diff
@@ -0,0 +1,420 @@
+        assert detail_response.data['deployment__active'] is True
+
+    def test_project_manager_can_archive_project(self):
+        manager = User.objects.get(username='anotheruser')
+        asset = Asset.objects.create(owner=User.objects.get(username='someuser'))
+        asset.deploy(backend='mock', active=True)
+        asset.assign_perm(manager, PERM_MANAGE_ASSET)
+        self._login_user('anotheruser')
+        response = self._create_send_payload([asset.uid], 'archive')
+        assert response.status_code == status.HTTP_200_OK
+
+        asset.refresh_from_db()
+        assert asset.deployment.active is False
+        assert asset.pending_delete is False
+
+        detail_response = self._get_asset_detail_results(asset.uid)
+        assert detail_response.status_code == status.HTTP_200_OK
+        assert detail_response.data['deployment__active'] is False
+
+    def test_superuser_can_undelete(self):
+        self._login_user('someuser')
+        asset = self._add_one_asset_for_someuser()
+        asset_uid = asset.uid
+        response = self._create_send_payload([asset_uid], 'delete')
+        assert response.status_code == status.HTTP_200_OK
+
+        asset.refresh_from_db()
+        assert asset.deployment.active is False
+        assert asset.pending_delete is True
+
+        # someuser cannot access their project anymore
+        detail_response = self._get_asset_detail_results(asset_uid)
+        assert detail_response.status_code == status.HTTP_404_NOT_FOUND
+
+        # superuser can undelete someuser's project
+        self._login_superuser()
+        response = self._create_send_payload([asset_uid], 'undelete')
+        assert response.status_code == status.HTTP_200_OK
+
+        asset.refresh_from_db()
+        assert asset.deployment.active is True
+        assert asset.pending_delete is False
+
+        # someuser can access their project again
+        detail_response = self._get_asset_detail_results(asset_uid)
+        assert detail_response.status_code == status.HTTP_200_OK
+        assert detail_response.data['deployment__active'] is True
+
+    def test_users_cannot_undelete(self):
+        self._login_user('someuser')
+        asset = self._add_one_asset_for_someuser()
+        asset_uid = asset.uid
+        response = self._create_send_payload([asset_uid], 'delete')
+        assert response.status_code == status.HTTP_200_OK
+
+        asset = Asset.all_objects.get(uid=asset_uid)
+        assert asset.deployment.active is False
+        assert asset.pending_delete is True
+
+        # someuser cannot access their project anymore
+        detail_response = self._get_asset_detail_results(asset_uid)
+        assert detail_response.status_code == status.HTTP_404_NOT_FOUND
+
+        # someuser is not allowed to undelete their project
+        response = self._create_send_payload([asset_uid], 'undelete')
+        assert response.status_code == status.HTTP_403_FORBIDDEN
```

Comment:
we discussed that if you pass `foo`, which could never exist, as an asset UID, you still get a 403 response. therefore, we're not leaking information about the existence of any particular UID :heavy_check_mark: